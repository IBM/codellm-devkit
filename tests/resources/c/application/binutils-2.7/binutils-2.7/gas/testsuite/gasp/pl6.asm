	.ALTERNATE
! test of expression operator
define 	MACRO 	val, string
	SDATA 	val
	SDATA	string
	ENDM
	define "1","99%of100"	! notice % within string
	define 	%1 + 2, "=3"


	define 	% 1 + 2 %3+4

	define	%3*4-2  <=10>

	define	%3*4-2  5

	define	%1 + 2,<is equal to %1 + 2, right?>

	! has no effect

	end
