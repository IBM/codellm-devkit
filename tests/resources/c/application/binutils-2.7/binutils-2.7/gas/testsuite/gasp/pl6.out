!	.ALTERNATE
!! test of expression operator
	! test of expression operator
!define 	MACRO 	val, string
!	SDATA 	val
!	SDATA	string
!	ENDM
!	define "1","99%of100"	! notice % within string
!	SDATA 	"1"
	.byte	49
!	SDATA	"99%of100"
	.byte	57,57,37,111,102,49,48,48
!	define 	%1 + 2, "=3"
!	SDATA 	3
	.byte	51
!	SDATA	"=3"
	.byte	61,51
!

!

!	define 	% 1 + 2 %3+4
!	SDATA 	3
	.byte	51
!	SDATA	7
	.byte	55
!

!	define	%3*4-2  <=10>
!	SDATA 	10
	.byte	49,48
!	SDATA	"=10"
	.byte	61,49,48
!

!	define	%3*4-2  5
!	SDATA 	10
	.byte	49,48
!	SDATA	5
	.byte	53
!

!	define	%1 + 2,<is equal to %1 + 2, right?>
!	SDATA 	3
	.byte	51
!	SDATA	"is equal to %1 + 2, right?"
	.byte	105,115,32,101,113,117,97,108,32,116,111,32,37,49,32,43,32,50,44,32,114,105,103,104,116,63
!

!	! has no effect
	! has no effect
!

!	end
